features/add-oembed-provider.feature

```diff
@@ -0,0 +1,314 @@
+Feature: Register and use custom oEmbed providers
+
+  Background:
+    Given a WP install
+    And a filter-providers.php file:
+      """
+      <?php
+      WP_CLI::add_wp_hook( 'init', function () {
+        wp_oembed_add_provider( 'http://example.com/*', 'http://example.com/api/oembed.{format}' );
+      });
+      """
+
+  Scenario: List oEmbed providers after registration
+    When I run `wp embed provider list --require=filter-providers.php`
+    And save STDOUT as {DEFAULT_STDOUT}
+    Then STDOUT should contain:
+      """
+      format
+      """
+    And STDOUT should contain:
+      """
+      endpoint
+      """
+    And STDOUT should not contain:
+      """
+      regex
+      """
+    And STDOUT should contain:
+      """
+      http://example.com/*
+      """
+    And STDOUT should contain:
+      """
+      http://example.com/api/oembed.{format}
+      """
+    And STDOUT should contain:
+      """
+      youtube\.com/watch.*
+      """
+    And STDOUT should contain:
+      """
+      //www.youtube.com/oembed
+      """
+    And STDOUT should contain:
+      """
+      flickr\.com/
+      """
+    And STDOUT should contain:
+      """
+      flickr.com
+      """
+    And STDOUT should contain:
+      """
+      twitter\.com/
+      """
+    And STDOUT should contain:
+      """
+      twitter.com
+      """
+    And STDOUT should contain:
+      """
+      \.spotify\.com/
+      """
+    And STDOUT should contain:
+      """
+      spotify.com
+      """
+
+    When I run `wp embed provider list --fields=format,endpoint --require=filter-providers.php`
+    Then STDOUT should be:
+      """
+      {DEFAULT_STDOUT}
+      """
+
+    When I run `wp embed provider list --force-regex --require=filter-providers.php`
+    Then STDOUT should contain:
+      """
+      #https?\://example\.com/(.+)#i
+      """
+    And STDOUT should contain:
+      """
+      http://example.com/api/oembed.{format}
+      """
+    And STDOUT should not contain:
+      """
+      http://example.com/*
+      """
+    And STDOUT should match /^#http/m
+
+    When I run `wp embed provider list --fields=format`
+    Then STDOUT should contain:
+      """
+      format
+      """
+    And STDOUT should not contain:
+      """
+      endpoint
+      """
+    And STDOUT should contain:
+      """
+      youtube\.com/watch.*
+      """
+    And STDOUT should not contain:
+      """
+      youtube.com
+      """
+
+    When I run `wp embed provider list --field=format`
+    Then STDOUT should not contain:
+      """
+      format
+      """
+    And STDOUT should not contain:
+      """
+      endpoint
+      """
+    And STDOUT should contain:
+      """
+      youtube\.com/watch.*
+      """
+    And STDOUT should not contain:
+      """
+      youtube.com
+      """
+
+    When I run `wp embed provider list --field=regex`
+    Then STDOUT should match /^(?:(?:1|0)\n)+$/
+
+  @require-wp-4.0
+  Scenario: Match an oEmbed provider
```

Comment:
Ditto here, how does this scenario test the new code? audio.com and YouTube aren't really new providers, no?